fortls/langserver.py

```diff
@@ -1101,6 +1101,11 @@ def create_signature_hover():
             elif var_obj.desc.endswith("STRING"):
                 hover_str = f"CHARACTER(LEN={len(var_obj.name)-2})"
                 hover_array.append(create_hover(hover_str, True))
+        else:
+            # Try and see if var_obj has any hover info. If it has, show it.
+            hover_str, highlight = var_obj.get_hover(long=True)
+            if hover_str:
+                hover_array.append(create_hover(hover_str, highlight))
```

Comment:
I missed this on my review. Is this the best way for this to be done? The doc parser cannot be very robust due to their dual (FORD/Doxygen) nature so it is bound to assign incorrect documentation in multiple AST nodes. I would explicitly check of a `var_obj` is `CLASS_TYPE_ID` or `MODULE_TYPE_ID`.

Also, if it's a `SUBMODULE_TYPE_ID` how should the documentation be parsed and assigned?